operations/kubernetes/Determine_if_Pods_are_Hitting_Resource_Limits.md

```diff
@@ -1,135 +1,98 @@
 
-1.  Use the detect\_cpu\_throttling.sh script to determine if any pods are being CP throttled.
+1. Use the `/opt/cray/platform-utils/detect_cpu_throttling.sh` script to determine if any pods are being CPU throttled _(this script is installed on master and worker NCNs)_.  The script can be used in two different ways:
 
-    1.  Install the script.
+    1. (`ncn-mw#`) Pass in a substring of the desired pod names.
 
-        The script can be installed on `ncn-w001`, or any NCN that can SSH to worker nodes.
+       In this example, the `externaldns` pods are being used:
 
-        ```bash
-        cat detect_cpu_throttling.sh
-        ```
-        
-        ```bash
-        #!/bin/sh
-        # Usage: detect_cpu_throttling.sh [pod_name_substr] (default evaluates all pods)
+       ```bash
+       /opt/cray/platform-utils/detect_cpu_throttling.sh externaldns
+       ```
 
-        str=$1
-        : ${str:=.}
+       Example output:
 
-        while read ns pod node; do
-          echo ""
-          echo "Checking $pod"
-          while read -r container; do
-            uid=$(echo $container | awk 'BEGIN { FS = "/" } ; {print $NF}')
-            ssh -T ${node} <<-EOF
-                dir=$(find /sys/fs/cgroup/cpu,cpuacct/kubepods/burstable -name *${uid}* 2>/dev/null)
-                [ "${dir}" = "" ] && { dir=$(find /sys/fs/cgroup/cpu,cpuacct/system.slice/containerd.service -name *${uid}* 2>/dev/null); }
-                if [ "${dir}" != "" ]; then
-                  num_periods=$(grep nr_throttled ${dir}/cpu.stat | awk '{print $NF}')
-                  if [ ${num_periods} -gt 0 ]; then
-                    echo "*** CPU throttling for containerid ${uid}: ***"
-                    cat ${dir}/cpu.stat
-                    echo ""
-                  fi
-                fi
-        	EOF
-          done <<< "`kubectl -n $ns get pod $pod -o yaml | grep ' - containerID'`"
-        done <<<"$(kubectl get pods -A -o wide | grep $str | grep Running | awk '{print $1 " " $2 " " $8}')"
-        ```
+       ```text
+       Checking cray-externaldns-coredns-58b5f8494-c45kh
 
-    2.  Determine if any pods are being CPU throttled.
+       Checking cray-externaldns-coredns-58b5f8494-pjvz6
 
-        The script can be used in two different ways:
+       Checking cray-externaldns-etcd-2kn7w6gnsx
 
-    -   Pass in a substring of the desired pod name\(s\).
+       Checking cray-externaldns-etcd-88x4drpv27
 
-        In the example below, the externaldns pods are being used.
+       Checking cray-externaldns-etcd-sbnbph52vh
 
-        ```bash
-        ./detect_cpu_throttling.sh externaldns
-        ```
+       Checking cray-externaldns-external-dns-5bb8765896-w87wb
+       *** CPU throttling: ***
+       nr_periods 1127304
+       nr_throttled 473554
+       throttled_time 71962850825439
+       ```
 
-        Example output:
+    1. (`ncn-mw#`) Call the script without a parameter to evaluate all pods.  It can take two minutes or more to run when evaluating all pods:
```

Comment:
```suggestion
    1. (`ncn-mw#`) Call the script without a parameter to evaluate all pods.

       This can take two minutes or more to run when evaluating all pods:
```